CharlieBackend.Data/Repositories/Impl/LessonRepository.cs

```diff
@@ -26,6 +26,24 @@ public new async Task<List<Lesson>> GetAllAsync()
                     .ToListAsync();
         }
 
+        public Task<Lesson> GetLastLesson()
+        {
+            return _applicationContext.Lessons.OrderByDescending(l => l.LessonDate).FirstOrDefaultAsync();
+        }
+
+        public Task<List<Lesson>> GetLessonsByDate(DateTime? startDate, DateTime? finishDate, long daysWeek, DateTime lastLessonDate)
```

Comment:
Please remove dayWeek and lastlessonDate parameters and implement that logic in the LessonService.GetLessonsByDate method.